rpc/namenode.go

```diff
@@ -145,25 +145,32 @@ func (c *NamenodeConnection) resolveConnection() error {
 	}
 
 	var err error
+	stackErrors := fmt.Errorf("resolveConnection failed. ")
 
 	if c.host != nil {
 		err = c.host.lastError
 	}
 
-	for _, host := range c.hostList {
+	for i, host := range c.hostList {
 		if host.lastErrorAt.After(time.Now().Add(-backoffDuration)) {
```

Comment:
Is this a rate limiting thing? It doesn't really seem like retries are done anywhere in the code to reconnect, so this is just preventing client retrys from pushing too hard on the host, I guess? I just skimmed the code so maybe I missed something.

Reply:
Since the connection is shared over all operations performed by the same client we can have a lot of concurrent dials to the same host. The backoff prevents a storm of network activity to a host with which we've previously failed to establish a connection.